### ISSUE TITLE
division by 0 in fuzzy_join when there are only perfect matches

### ISSUE THREAD

**Issue by jeromedockes** [skrub-data/skrub#764]
### Describe the bug

if we join 2 tables on columns for which matches do exist, the rescaling of nearest neighbor distances does a division by 0; we get a warning and the matching scores are NaN

### Steps/Code to Reproduce

```python
>>> warnings.filterwarnings("default")
>>> import pandas as pd
... from skrub import fuzzy_join
... 
... df1 = pd.DataFrame({"A": [0, 1]})
... 
... fuzzy_join(df1, df1, on="A", return_score=True)
/home/jerome/workspace/backedup_repositories/skrub/skrub/_fuzzy_join.py:359: UserWarning: This feature is still experimental.
  warnings.warn("This feature is still experimental.")
/home/jerome/workspace/backedup_repositories/skrub/skrub/_fuzzy_join.py:189: RuntimeWarning: invalid value encountered in divide
  distance = distance / np.max(distance)
   A_x  A_y  matching_score
0    0    0             NaN
1    1    1             NaN
>>> 
```

### Expected Results

no division by 0, scores are numbers (prob. equal to 1)

### Actual Results

division by 0, scores are NaN

### Versions

```shell
main branch
```

**Comment by GaelVaroquaux** [comment]
Good catch, this is an important special case.

Ideally, when there are only exact matches, we shouldn't even go down that code path and only use standard matching from the backend (pandas, polars...), as it will be much faster.

**Comment by jovan-stojanovic** [comment]
Yes, indeed, you are right. This is IMO equivalent to #587 if we agree that we should rely on the backend for these matches..

**Comment by jeromedockes** [comment]
this is not the same as when the user requires perfect matches by setting `match_score=1`. This division by 0 is triggered when all rows happen to have a perfect match in the joined dataframe. it will be fixed by #802

**Comment by jeromedockes** [comment]
fixed by #802
